sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/SonarAnalysisContext.cs

```diff
@@ -129,15 +137,28 @@ internal void RegisterCodeBlockStartAction<TLanguageKindEnum>(Action<CodeBlockSt
             context.RegisterSymbolAction(
                 c =>
                 {
-                    if (!IsAnalysisDisabled(c.Symbol.Locations.FirstOrDefault(l => l.SourceTree != null)?.SourceTree))
+                    if (!IsAnalysisDisabled(c.Symbol.Locations.FirstOrDefault(l => l.SourceTree != null)?.SourceTree,
+                            supportedDiagnostics))
                     {
                         action(c);
                     }
                 }, symbolKinds);
 
-        internal static bool IsAnalysisDisabled(SyntaxTree tree) =>
-            tree != null &&
-            ShouldAnalysisBeDisabled != null &&
-            ShouldAnalysisBeDisabled(tree);
+        internal static bool IsAnalysisDisabled(SyntaxTree tree, IEnumerable<DiagnosticDescriptor> supportedDiagnostics)

```

Comment:
Can we change it to positive? For example `IsAnalysisEnabled` or `IsAnalysisEnabledByFoo` (to highlight it could be enabled/disabled by multiple mechanisms).

Reply:
It doesn't matter much as it will be refactored by the following tickets (same sprint).